rasa/cli/x.py

```diff
@@ -112,10 +112,24 @@ def _overwrite_endpoints_for_local_x(
     from rasa.utils.endpoints import EndpointConfig
     import questionary
 
+    # Checking if endpoint.yml has existing url and wait time values set, if so give warning we are overwriting
+    # the endpoint.yml file.
+    custom_wait_time_pulls = endpoints.model.kwargs.get("wait_time_between_pulls")
+    custom_url = endpoints.model.url
+    default_rasax_model_server_url = (
+        f"{rasa_x_url}/projects/default/models/tag/production"
+    )
+
+    if custom_url != default_rasax_model_server_url:
+        logger.info(
```

Comment:
Mhm, I think a `warnings.warn` would be even better here than `info` (a warning because we overwrite things, and `warnings.warn` because the user can avoid this warning by not specifying a wrong url here). Wdyt?

Reply:
yea I think that is the right approach as well, updating it

Reply:
Hey @wochinge do you know why this wouldn't work?  I am getting the warning message ok from testing but this function doesn't seem to find it:

```
    with assert_log_emitted(caplog, "rasa.cli.x", warnings.warn, log_message):
        x._overwrite_endpoints_for_local_x(endpoints_custom, "test", "http://localhost")
```

I thought this would work by just updating the warnings.warn or even tried logger.warn and still get the below error on tests:

```
E           AssertionError: Did not detect expected logging output.
E           Expected output is (logger name, log level, text): (rasa.cli.x, <built-in function warn>, Ignoring url 'http://testserver:5002/models/default@latest' from 'endpoints.yml' and using 'http://localhost/projects/default/models/tag/production' instead)
E           Instead found records:
E           []
```

I'm working through it just thought I'd ask

Reply:
looks like this resolved that issue, is this the proper way to do this?

```
    with warnings.catch_warnings(record=True):
        x._overwrite_endpoints_for_local_x(endpoints_custom, "test", "http://localhost")
```

Reply:
I'd use this one: 

https://docs.pytest.org/en/latest/warnings.html#warns

Reply:
we just recently (Nikolai merged it this weekend) switched to `warnings` so we are all a bit new to it 🙂

Reply:
ok cool thanks for the info I'll get that going.